executor/src/witgen/global_constraints.rs

```diff
@@ -254,16 +249,26 @@ fn propagate_constraints<T: FieldElement>(
                 false
             }
         }
-        Identity::Lookup(LookupIdentity { left, right, .. })
-        | Identity::PhantomLookup(PhantomLookupIdentity { left, right, .. }) => {
-            if !left.selector.is_one() || !right.selector.is_one() {
+        // Bus receives are handled when they are matched with a send.
+        Identity::BusReceive(..) => false,
+        Identity::BusSend(send) => {
+            let receive = send.try_match_static(bus_receives).unwrap();
+            if !receive.is_unconstrained() {
+                return false;
+            }
+            if !send.selected_tuple.selector.is_one() || !receive.selected_tuple.selector.is_one() {
```

Comment:
If the send selector is constant one but the receive one is not, we can still transfer range constraints to the send data, can't we?

Reply:
Works